Player#sendEquipmentChange produces invalid packet with empty map (causes disconnect)

### Expected behavior

Method call should silently fail

### Observed/Actual behavior

ClientboundSetEquipmentPacket is sent anyway

### Steps/models to reproduce

invoke Player#sendEquipmentChange with an empty map as slots argument

### Plugin and Datapack List

irrelevant

### Paper version

[21:30:36 INFO]: Checking version, please wait...
[21:30:36 INFO]: This server is running Paper version 1.21.4-121-main@88bbead (2025-01-22T20:05:15Z) (Implementing API version 1.21.4-R0.1-SNAPSHOT)
You are running the latest version
Previous version: 1.21.4-118-73f7736 (MC: 1.21.4)


### Other

ClientboundSetEquipmentPacket does not allow for empty slots

    io.netty.handler.codec.DecoderException: Failed to decode packet 'clientbound/minecraft:set_equipment'

    Caused by: java.lang.IndexOutOfBoundsException: readerIndex(2) + length(1) exceeds writerIndex(2): PooledUnsafeDirectByteBuf(ridx: 2, widx: 2, cap: 2)

--- a/paper-api/src/main/java/org/bukkit/entity/Player.java
+++ b/paper-api/src/main/java/org/bukkit/entity/Player.java
@@ -1063,7 +1063,7 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
      *
      * @param entity the entity whose equipment to change
      * @param items the slots to change, where the values are the items to which
-     * the slot should be changed. null values will set the slot to air
+     * the slot should be changed. null values will set the slot to air, empty map is not allowed
      */
     public void sendEquipmentChange(@NotNull LivingEntity entity, @NotNull Map<EquipmentSlot, ItemStack> items);
 

--- a/paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1161,6 +1161,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
     public void sendEquipmentChange(LivingEntity entity, Map<EquipmentSlot, ItemStack> items) {
         Preconditions.checkArgument(entity != null, "Entity cannot be null");
         Preconditions.checkArgument(items != null, "items cannot be null");
+        Preconditions.checkArgument(!items.isEmpty(), "items cannot be empty");
 
         if (this.getHandle().connection == null) {
             return;

